tools/profcheck/check.go

```diff
@@ -0,0 +1,363 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	profiles "go.opentelemetry.io/proto/otlp/profiles/v1development"
+	"google.golang.org/protobuf/proto"
+)
+
+func CheckConformance(data *profiles.ProfilesData) error {
+	dict := data.Dictionary
+	if len(data.ResourceProfiles) == 0 {
+		return errors.New("resource profiles are empty")
+	}
+	for _, resourceProfiles := range data.ResourceProfiles {
+		// TODO: Check attributes?
+		for _, scopeProfiles := range resourceProfiles.ScopeProfiles {
+			// TODO: Check attributes?
```

Comment:
As Attributes in ScopeProfiles are not part of the ProfilesDictionary, I think we don't need to check them.